manager/util/tokenrequest.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2021 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	cmclient "github.com/jetstack/cert-manager/pkg/client/clientset/versioned"
+	"k8s.io/client-go/rest"
+
+	"github.com/cert-manager/csi-lib/manager"
+	"github.com/cert-manager/csi-lib/metadata"
+)
+
+// ClientForMetadataTokenRequestEmptyAud returns a
+// manager.ClientForMetadataFunc that returns a cert-manager rest client whose
+// authentication is built using the passed empty audience ("") token request
```

Comment:
Suggestion: I understand that the empty audience is so that the audience defaults to kube API server. Would it make sense to add that to the comment or is it unlikely that a user of the function would not already know that?

Reply:
I've tried to add this context in the following sentence:

> The resulting cert-manager client is
// authenticated against the Kubernetes API server using the mounting Pod's
// ServiceAccount.

It is not that it _defaults_ to the kube API server, it _is_ the kube API server audience, [see](https://kubernetes-csi.github.io/docs/token-requests.html#usage).

Reply:
Yeah makes sense especially since it's in the CSI dev docs too which I didn't realize 👍🏼